test/test_000_build_containers.py

```diff
@@ -6,10 +6,15 @@
     "local://"
 ).get_module("Command").run
 
+
 @pytest.mark.parametrize("image,tag", [
-    ( 'test/debian.Dockerfile', 'pytest_pihole:debian' ),
-    ( 'test/centos.Dockerfile', 'pytest_pihole:centos' ),
+    ('test/debian.Dockerfile', 'pytest_pihole:debian'),
+    ('test/centos.Dockerfile', 'pytest_pihole:centos'),
+    ('test/fedora.Dockerfile', 'pytest_pihole:fedora'),
 ])
+# mark as 'build_stage' so we can ensure images are build first when tests
+# are executed in parallel. (not required when tests are executed serially)
+@pytest.mark.build_stage
```

Comment:
We still want the files to be ordered for when the tests are executed with simple `py.test -vv`. Lexical sorting is a good thing. I actually made this `mark` up.. so we can call it whatever we want instead of `build_stage`.

Reply:
ah good point I forgot about.  I had the same conundrum with docker-pi-hole and ended up just adding a seperate script [Dockerfile.py](https://github.com/diginc/docker-pi-hole/blob/master/Dockerfile.py) (also template my multiple architectures) to do the build process and then glued it back together with the tests using [tox.ini](https://github.com/diginc/docker-pi-hole/blob/master/tox.ini) to run the docker image build followed by pytest.